operators/src/processing/time_projection/mod.rs

```diff
@@ -0,0 +1,560 @@
+                step_reference: self.step_reference,
+            }.boxed().into()),
+        )
+    }
+}
+
+pub struct VectorTimeProjectionProcessor<G>
+where
+    G: Geometry,
+{
+    processor: Box<dyn VectorQueryProcessor<VectorType = FeatureCollection<G>>>,
+    step: TimeStep,
+    step_reference: TimeInstance,
+}
+
+#[async_trait]
+impl<G> VectorQueryProcessor for VectorTimeProjectionProcessor<G>
+where
+    G: Geometry + ArrowTyped + 'static,
+{
+    type VectorType = FeatureCollection<G>;
+
+    async fn vector_query<'a>(
+        &'a self,
+        query: VectorQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<BoxStream<'a, Result<Self::VectorType>>> {
+        let query = self.expand_query_rectangle(query)?;
+        let stream = self
+            .processor
+            .vector_query(query, ctx)
+            .await?
+            .and_then(|collection| {
+                self.expand_feature_collection_result(collection, ctx.thread_pool().clone())
+            })
+            .boxed();
+        Ok(stream)
+    }
+}
+
+impl<G> VectorTimeProjectionProcessor<G>
+where
+    G: Geometry + ArrowTyped + 'static,
+{
+    fn expand_query_rectangle(&self, query: VectorQueryRectangle) -> Result<VectorQueryRectangle> {
+        Ok(expand_query_rectangle(
+            self.step,
+            self.step_reference,
+            query,
+        )?)
+    }
+
+    async fn expand_feature_collection_result(
+        &self,
+        feature_collection: FeatureCollection<G>,
+        thread_pool: Arc<ThreadPool>,
+    ) -> Result<FeatureCollection<G>> {
+        let step = self.step;
+        let step_reference = self.step_reference;
+
+        crate::util::spawn_blocking_with_thread_pool(thread_pool, move || {
+            Self::expand_feature_collection_result_inner(feature_collection, step, step_reference)
+        })
+        .await?
+        .map_err(Into::into)
+    }
+
+    fn expand_feature_collection_result_inner(
+        feature_collection: FeatureCollection<G>,
+        step: TimeStep,
+        step_reference: TimeInstance,
+    ) -> Result<FeatureCollection<G>, TimeProjectionError> {
+        let time_intervals: Option<Result<Vec<TimeInterval>, TimeProjectionError>> =
+            feature_collection
+                .time_intervals()
+                .par_iter()
+                .with_min_len(128) // TODO: find good default
+                .map(|time_interval| expand_time_interval(step, step_reference, *time_interval))
+                // TODO: change to [`try_collect_into_vec`](https://github.com/rayon-rs/rayon/issues/713) if available
+                .try_fold_with(Vec::new(), |mut acc, time_interval| {
+                    acc.push(time_interval?);
+                    Ok(acc)
+                })
+                .try_reduce_with(|a, b| Ok([a, b].concat()));
+
+        match time_intervals {
+            Some(Ok(time_intervals)) => feature_collection
+                .replace_time(&time_intervals)
```

Comment:
could we just add a new column as _the_ time column and keep the old information around? maybe make that a config option?

Reply:
We have to wait for the new timestamp column type first.